core/include/detail/backends/win32_edge.hh

```diff
@@ -1,909 +1,692 @@
+    }
+    return error_info{WEBVIEW_ERROR_INVALID_STATE};
+  }
+  noresult terminate_impl() override {
+    PostQuitMessage(0);
+    return {};
+  }
+  noresult dispatch_impl(dispatch_fn_t f) override {
+    PostMessageW(m_message_window, WM_APP, 0, (LPARAM) new dispatch_fn_t(f));
+    // warning: Potential memory leak [clang-analyzer-cplusplus.NewDeleteLeaks]
+    return {};
+  }
+
+  noresult set_title_impl(const std::string &title) override {
+    SetWindowTextW(m_window, win_string::widen_string(title).c_str());
+    return {};
+  }
+
+  noresult set_size_impl(int width, int height, webview_hint_t hints) override {
+    auto style = GetWindowLong(m_window, GWL_STYLE);
+    if (hints == WEBVIEW_HINT_FIXED) {
+      style &= ~(WS_THICKFRAME | WS_MAXIMIZEBOX);
+    } else {
+      style |= (WS_THICKFRAME | WS_MAXIMIZEBOX);
+    }
+    SetWindowLong(m_window, GWL_STYLE, style);
+
+    if (hints == WEBVIEW_HINT_MAX) {
+      m_maxsz.x = width;
+      m_maxsz.y = height;
+    } else if (hints == WEBVIEW_HINT_MIN) {
+      m_minsz.x = width;
+      m_minsz.y = height;
+    } else {
+      auto dpi = dpi::get_window_dpi(m_window);
+      m_dpi = dpi;
+      auto scaled_size =
+          dpi::scale_size(width, height, dpi::get_default_window_dpi(), dpi);
+      auto frame_size = dpi::make_window_frame_size(m_window, scaled_size.cx,
+                                                    scaled_size.cy, dpi);
+      SetWindowPos(m_window, nullptr, 0, 0, frame_size.cx, frame_size.cy,
+                   SWP_NOZORDER | SWP_NOACTIVATE | SWP_NOMOVE |
+                       SWP_FRAMECHANGED);
+    }
+    return window_show();
+  }
+
+  noresult navigate_impl(const std::string &url) override {
+    auto wurl = win_string::widen_string(url);
+    m_webview->Navigate(wurl.c_str());
+    return {};
+  }
+
+  noresult eval_impl(const std::string &js) override {
+    // TODO: Skip if no content has begun loading yet. Can't check with
+    //       ICoreWebView2::get_Source because it returns "about:blank".
+    auto wjs = win_string::widen_string(js);
+    m_webview->ExecuteScript(wjs.c_str(), nullptr);
+    return {};
+  }
+
+  noresult set_html_impl(const std::string &html) override {
+    m_webview->NavigateToString(win_string::widen_string(html).c_str());
+    return {};
+  }
+
+  user_script add_user_script_impl(const std::string &js) override {
+    auto wjs = win_string::widen_string(js);
+    std::wstring script_id;
+    bool done{};
+    webview2_user_script_added_handler handler{[&](HRESULT res, LPCWSTR id) {
+      if (SUCCEEDED(res)) {
+        script_id = id;
+        // We need a reload, else the user script will never execute unless the user explicitly calls
+        // `navigate` after `init`
+        if (owns_window()) {
+          reload_browser_window();
+        }
+      } else {
+        console.error("Windows failed to add user script.",
+                      static_cast<webview_error_t>(res));
+      }
+      done = true;
+    }};

```

Comment:
With reference to #1255

@SteffenL and @AlbertXingZhang, as far as Heisenburg bugs are concerned, I did manage to observe this error log being triggered once during a CI tests run. Re-running the failed test subsequently succeeded.

If not for the error log, the only symptom was that the windows test timed out (which is the mystery symptom...).
My memory is 98% clear on this.